packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/container.tsx

```diff
@@ -44,11 +41,17 @@ const ResetPasswordWithRecoveryKeyVerifiedContainer = ({
     useState<FinishOAuthFlowHandlerResult['error']>();
 
   const account = useAccount();
-  const { uid, sessionToken } = currentAccount() || {};
+  // TODO: how do we properly handle the real non-null nature?
+  const { uid, sessionToken, email = '💩' } = currentAccount() || {};
```

Comment:
💩 🤣 

`currentAccount` pulls from local storage right, which is what we want most of the time 👍 but just from a quick search, I only see us call `storeAccountData` in `CompleteResetPassword` so is this available from local storage at this point in the flow...? If not, then location state seems like maybe the correct place?

To hopefully answer the code comment question, _before_ the `return (<ResetPasswordWithRecoveryKeyVerified` can we just check that these values are non-null and if so, redirect? See how we check for `!email` in the signin container component. (I don't think this is applicable here but if we ever have a case where we're passing non-null values into a component just to pass them back to a callback, we can curry them, see `InlineRecoveryKeySetup` and `SetPassword` containers)

FWIW related to your PR title, we shouldn't be using `useAccount` unless we want the entire GQL query to run, we've got [a ticket to remove it in reset PW](https://mozilla-hub.atlassian.net/browse/FXA-9672).

Reply:
This page is reached after `CompleteResetPassword` so email should be stored in local storage at this stage - if it isn't, that means the user isn't signed in and should likely be redirected to `/signin`

Reply:
Ah thanks Valerie, I can never keep these reset PW flows straight. So yeah, `currentAccount` is the right thing here and I think if the value is null we can just redirect to `/signin` as you said.